cgi-bin/DW/Controller/EditIcons.pm

```diff
@@ -0,0 +1,391 @@
+        if ($POST->{"kw_$picid"} ne $POST->{"kw_orig_$picid"}) {
+            my $kws = $POST->{"kw_$picid"};
+
+            if ( $POST->{"rename_keyword_$picid"} ) {
+                if ( $display_rename && $u->userpic_have_mapid ) {
+                    eval {
+                       $up->set_and_rename_keywords($kws, $POST->{"kw_orig_$picid"});
+                    } or push @$errors, $@->as_html;
+                } else {
+                    push @$errors, $ML->{'.label.rename.disabled'};
+                }
+            } else {
+                eval {
+                    $up->set_keywords($kws);
+                } or push @$errors, $@->as_html;
+            }
+        }
+
+        eval {
+            $up->set_comment ($POST->{"com_$picid"})
+                unless $POST->{"com_$picid"} eq $POST->{"com_orig_$picid"};
+        } or push @$errors, $@;
+
+        eval {
+            $up->set_description ($POST->{"desc_$picid"})
+                unless $POST->{"desc_$picid"} eq $POST->{"desc_orig_$picid"};
+        } or push @$errors, $@;
+
+    }
+
+    foreach my $kw (keys %used_keywords) {
+        next unless $used_keywords{$kw} > 1;
+        push @$errors, BML::ml('.error.keywords', {ekw => $kw});
+    }
+
+    if (@delete && $LJ::DISABLE_MEDIA_UPLOADS) {
+        push @$errors, $ML->{'.error.nomediauploads.delete'};
+
+    } elsif (@delete) {
+
+        # delete pics
+        foreach my $up (@delete) {
+            eval { $up->delete; } or push @$errors, $@;
+        }
+
+        # if any of the userpics they want to delete are active, then we want to
+        # re-run activate_userpics() - turns out it's faster to not check to
+        # see if we need to do this
+        $u->activate_userpics;
+    }
+
+    my $new_default = $POST->{'defaultpic'}+0;
+    if ($POST->{"delete_${new_default}"}) {
+        # deleting default
+        $new_default = 0;
+    }
+
+    if ($new_default && $new_default != $u->{'defaultpicid'}) {
+        my ($up) = grep { $_->id == $new_default } @userpics;
+
+        # see if they are trying to make an inactive userpic their default
+        if ($up && !$up->inactive) {
+            $up->make_default;
+        }
+    } elsif ($new_default eq '0' && $u->{'defaultpicid'}) {
+        # selected the "no default picture" option
+        $u->update_self( { defaultpicid => 0 } );
+        $u->{'defaultpicid'} = 0;
+    }
+
+    return scalar @delete;
+}
+
+sub parse_post_uploads {
+    my ( $ML, $POST, $u, $MAX_UPLOAD ) = @_;
+    my @uploads;
+
+    # if we find a userpic that requires the factory, save it here.
+    # we can only have one.
+    my $requires_factory;
+
+    # go through each key and create a %current_upload for it, then
+    # put it in @uploads.
+    foreach my $userpic_key ( keys %$POST ) {
+        my @tokens = split(/_/, $userpic_key);
+        my $counter = $tokens[1];
+
+        my %current_upload = (
+            image        => $POST->{$userpic_key},
```

Comment:
I... don't know. I've re-added it and will test. Okay. That breaks everything. I think removing it is necessary but I don't know why. Uh. It's probably an interaction with the difference in what BlobStore expects as the input? :|